packages/block-library/src/navigation/edit/index.js

```diff
@@ -297,6 +297,24 @@ function Navigation( {
 		[ clientId ]
 	);
 	const isResponsive = 'never' !== overlayMenu;
+
+	const shouldBeCollapsed = useCallback( () => {
+		return (
+			'always' === overlayMenu ||
+			( 'mobile' === overlayMenu && window.innerWidth < 600 )
+		);
+	}, [ overlayMenu ] );
+	const [ isCollapsed, setIsCollapsed ] = useState( shouldBeCollapsed() );
+	useEffect( () => {
+		// Update when the window resizes.
+		window.addEventListener( 'resize', () => {
+			setIsCollapsed( shouldBeCollapsed() );
+		} );
```

Comment:
I don't think you need state here. If you look at [what I did in my PR](https://github.com/WordPress/gutenberg/pull/45274/files) you will see you can use hooks for this:

```js
const isMobileBreakPoint = useMediaQuery(`(max-width: ${mobileBreakpoint})`);
```

You can adapt it to remove the event listener stuff and remove all this state and effects.

I appreciate you are utilising `shouldBeCollapsed` but you can still do this and simply conditionally apply the result of the `useMediaQuery` and rename `shouldBeCollapsed` to `shouldAlwaysBeCollapsed` and make it check for the `always` only. It shouldn't need to care about the initial window size.

Reply:
Done in c8f36c1513